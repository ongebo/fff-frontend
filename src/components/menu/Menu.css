/*---------------------------------------------
CONTENT STYLES
---------------------------------------------*/

.column {
  float: left;
}

.main {
  width: 60%;
}

.side {
  width: 40%;
}

.order-list {
  width: 90%;
  margin: auto;
  background: white;
  box-shadow: 0 0 30px lightgrey;
  padding: 0 10px 5px 10px;
  margin-top: 20px;
  margin-bottom: 20px;
  height: 100%;
  overflow: auto;
  text-align: center;
}

@media only screen and (max-width: 600px) {
  .column {
    float: none;
  }
  .main,
  .side,
  .order-list,
  .food-menu {
    width: 100%;
  }
}

@media only screen and (min-width: 600px) and (max-width: 1000px) {
  .order-list,
  .food-menu {
    width: 95%;
    box-shadow: none;
    border: 1px solid lightgrey;
  }
  .order-total,
  .title {
    font-size: 1em;
  }
}
