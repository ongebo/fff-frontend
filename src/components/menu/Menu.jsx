import React, { Component } from 'react';
import { connect } from 'react-redux';
import OrderList from './OrderList';
import MenuItem from './MenuItem';
import './Menu.css';
import { getFoodMenu } from '../../actions/FoodOrderActions';

export class Menu extends Component {
  constructor(props) {
    super(props);
    this.state = {
      orderList: []
    };

    this.addItemToList = this.addItemToList.bind(this);
    this.resetOrderList = this.resetOrderList.bind(this);
  }

  componentDidMount() {
    this.props.getFoodMenu(this.props.token);
  }

  addItemToList(item) {
    this.setState(state => {
      let inList = false;
      let orderList = [...state.orderList];
      for (let c = 0; c < orderList.length; c++) {
        if (item.item === orderList[c].item) {
          inList = true;
          orderList[c].quantity = item.quantity;
          orderList[c].cost = item.cost;
          break;
        }
      }
      if (!inList) {
        orderList.push(item);
      }
      return { orderList: orderList.filter(item => item.quantity != 0) };
    });
  }

  resetOrderList() {
    this.setState({ orderList: [] });
  }

  render() {
    return (
      <div className="clearfix">
        <div className="column main">
          <div className="food-menu">
            <h1 className="title">Food Menu</h1>
            {this.props.menu.map(item => (
              <MenuItem
                key={item.id}
                item={item.item}
                rate={item.rate}
                unit={item.unit}
                add={this.addItemToList}
              />
            ))}
          </div>
        </div>
        <OrderList
          orderList={this.state.orderList}
          resetOrderList={this.resetOrderList}
        />
      </div>
    );
  }
}

function mapStateToProps(state) {
  return { token: state.authentication.token, menu: state.menu };
}

export default connect(
  mapStateToProps,
  { getFoodMenu }
)(Menu);
