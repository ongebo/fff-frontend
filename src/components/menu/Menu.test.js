import React from 'react';
import Enzyme from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
import { Menu } from './Menu';

Enzyme.configure({ adapter: new EnzymeAdapter() });

describe('<Menu />', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(
      <Menu menu={[]} token={''} getFoodMenu={jest.fn()} />
    );
    expect(wrapper).toMatchSnapshot();
    const instance = wrapper.instance();
    instance.addItemToList({item: '', quantity: '', cost: ''});
  });

  it('contains a clearfix <div />', () => {
    const wrapper = shallow(
      <Menu menu={[]} token={''} getFoodMenu={jest.fn()} />
    );
    expect(wrapper.find('.clearfix')).toHaveLength(1);
  });
});
