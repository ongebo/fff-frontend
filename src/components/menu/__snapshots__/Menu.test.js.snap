// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> matches snapshot 1`] = `ShallowWrapper {}`;
